### PR TITLE
ENG-2376: make solana tx building work

### PR DESCRIPTION
**Pull request by jdomingos**
## Added

_Description of new functionality, feature, or content that has been added in this pull request._

## Changed

Updated the Solana connector to make transaction decoding deterministic and safer:

getSolanaTxDecodingCandidates() now chooses one encoding path:
hex only when the input is valid hex (including 0x...)
otherwise base64 only
This prevents hex payloads from also being (incorrectly) retried as base64.
Also tightened tests for this behavior:

Added assertion that a 0x hex input produces exactly one candidate, with encoding "hex".
Kept validation coverage for base64, error messaging, and adapter pass-through.
Verified with pnpm lint and the connector test file (all passing).

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced Solana transaction processing to robustly support multiple encoding formats (base64 and hexadecimal) with significantly improved error messages that provide better diagnostic context and clarity when transaction decoding fails.

* **Tests**
  * Added comprehensive test coverage validating transaction decoding across various payload formats, different encoding scenarios, and error handling edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stakekit/widget/pull/518?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/widget/src/providers/misc/solana-connector.ts [hunk 1]
```diff
@@ -25,6 +25,90 @@ import {
...
+    candidates.push({
+      encoding: "hex",
+      buffer: Buffer.from(withoutHexPrefix, "hex"),
+    });
+  }
+
+  candidates.push({
+    encoding: "base64",
+    buffer: Buffer.from(normalizedTx, "base64"),
+  });
```
[the remark this answers is not in the document]
Comment (jdomingos) [reply packages/widget/src/providers/misc/solana-connector.ts]: fixed

**Review by petar-omni — APPROVED** [review]

**Review by Philippoes — APPROVED** [review]
